src/App.tsx

```diff
@@ -15,47 +16,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'DEFAULT',
+  Alphabetically = 'ALPHABETICALLY',
+  Length = 'LENGTH',
+  Reverse = 'REVERSE',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState(SortType.Default);
+
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reset = () => {
+    setVisibleGoods([...goodsFromServer]);
+    setSortField(SortType.Default);
+    setIsReversed(false);
+  };
+
+  const prepareGoods = (type: SortType) => {
+    if (type === SortType.Reverse) {
+      setVisibleGoods([...visibleGoods].reverse());
+      return;
+    }
```

Comment:
The `prepareGoods` function reverses the `visibleGoods` array when `SortType.Reverse` is selected. However, this logic does not account for the current sort field. Consider integrating the reverse logic with the existing sorting logic to ensure that the list is reversed based on the current sort criteria.